From 1e1224df1fd8931d7547ea8a3df1c7154da32ad5 Mon Sep 17 00:00:00 2001
From: Tim Carney <timothy.carney@lairdtech.com>
Date: Fri, 4 Dec 2015 17:09:36 -0600
Subject: [PATCH] laird: dts: Add default SPIDEV device to device tree

This will serve as an example of how to add a SPI device to the
device tree as well as allow the customer to use the exisitng
SPIDEV driver to control the SPI lines with a userspace SPI
driver.
---
 arch/arm/boot/dts/wb50n.dts      | 6 ++++++
 arch/arm/configs/wb50n_defconfig | 1 +
 2 files changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/wb50n.dts b/arch/arm/boot/dts/wb50n.dts
index 04cdbf3..cd663ef 100644
--- a/arch/arm/boot/dts/wb50n.dts
+++ b/arch/arm/boot/dts/wb50n.dts
@@ -139,6 +139,12 @@
 			spi1: spi@f8008000 {
 				cs-gpios = <&pioC 25 0>, <0>, <0>, <0>;
 				status = "okay";
+
+			        spidev@0 {
+				          compatible = "spidev";
+				          reg = <0>;
+				          spi-max-frequency = <8000000>;
+			         };
 			};
 
 			watchdog@fffffe40 {
diff --git a/arch/arm/configs/wb50n_defconfig b/arch/arm/configs/wb50n_defconfig
index 1c27304..7a392f1 100644
--- a/arch/arm/configs/wb50n_defconfig
+++ b/arch/arm/configs/wb50n_defconfig
@@ -174,6 +174,7 @@ CONFIG_I2C_MUX=y
 CONFIG_I2C_AT91=y
 CONFIG_I2C_GPIO=y
 CONFIG_SPI=y
+CONFIG_SPI_SPIDEV=m
 CONFIG_SPI_ATMEL=m
 CONFIG_GPIO_SYSFS=y
 CONFIG_POWER_SUPPLY=y
-- 
1.9.1

